### PR TITLE
feat(buildpulse): Cross-platform GitHub Action

### PR DESCRIPTION
**Pull request by janpio**
Follow up to https://github.com/prisma/prisma/pull/12054

@jasonrudolph

### CODE REVIEW

**Comment by janpio** [comment]
That seems to be working fine @jasonrudolph.

**Comment by jasonrudolph** [comment]
> That seems to be working fine @jasonrudolph.

Great! I spot-checked the data on the @buildpulse back-end as well, and everything looks good. 👍 

Thanks for setting this up, @janpio. :bow:

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -723,16 +724,15 @@ jobs:
...
-        if: startsWith(github.repository, 'prisma/') && runner.os == 'macOS' && !cancelled()
-        env:
-          BUILDPULSE_ACCESS_KEY_ID: ${{ secrets.BUILDPULSE_ACCESS_KEY_ID }}
-          BUILDPULSE_SECRET_ACCESS_KEY: ${{ secrets.BUILDPULSE_SECRET_ACCESS_KEY }}
-        run: |
-          curl -fsSL https://get.buildpulse.io/test-reporter-darwin-amd64 > ./buildpulse-test-reporter
-          chmod +x ./buildpulse-test-reporter
-          ./buildpulse-test-reporter submit packages --account-id 17219288 --repository-id 192925833
+        if: env.HAS_BUILDPULSE_SECRETS == 'true' && !cancelled()
+        uses: Workshop64/buildpulse-action@cross-platform
```
Comment (jasonrudolph) [inline .github/workflows/test.yml:729]:
```suggestion
        uses: Workshop64/buildpulse-action@main
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -810,13 +810,12 @@ jobs:
...
-        if: startsWith(github.repository, 'prisma/') && runner.os == 'macOS' && !cancelled()
-        env:
-          BUILDPULSE_ACCESS_KEY_ID: ${{ secrets.BUILDPULSE_ACCESS_KEY_ID }}
-          BUILDPULSE_SECRET_ACCESS_KEY: ${{ secrets.BUILDPULSE_SECRET_ACCESS_KEY }}
-        run: |
-          curl -fsSL https://get.buildpulse.io/test-reporter-darwin-amd64 > ./buildpulse-test-reporter
-          chmod +x ./buildpulse-test-reporter
-          ./buildpulse-test-reporter submit packages/client --account-id 17219288 --repository-id 192925833
+        if: env.HAS_BUILDPULSE_SECRETS == 'true' && !cancelled()
+        uses: Workshop64/buildpulse-action@cross-platform
```
Comment (jasonrudolph) [inline .github/workflows/test.yml:815]:
```suggestion
        uses: Workshop64/buildpulse-action@main
```

**Review by jasonrudolph — COMMENTED** [review]
Now that https://github.com/Workshop64/buildpulse-action/pull/3 is merged, you can use the action's `main` branch instead of the temporary `cross-platform` branch, so I've suggested those edits in the review below.

I hope this helps.

**Review by Jolg42 — APPROVED** [review]

**Comment by Jolg42** [comment]
@janpio Can we merge this?

**Comment by janpio** [comment]
We have 3 failing tests, but they should not be influenced by this really, so if that is just generaly flakyness: Yes.

**Comment by Jolg42** [comment]
Yes, it's unrelated, then, merging now ✔️

**Pull request merged**